File: My Answer/E6.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

public class E6 {

    public static final Scanner scanner = new Scanner(System.in);

    public static void Q1(){
        Map<String, Double> clientes = new HashMap<>();
        clientes.put("Lira", 5000.00);
        clientes.put("Alon", 3000.00);
        clientes.put("Julia", 4500.00);
        System.out.println(clientes);
        String key = "Alon";
        if(clientes.containsKey(key)){
            clientes.put(key, clientes.get(key) + 1500.00);
            System.out.println(clientes);
        }
        clientes.put("Marcos", 2000.00);
        System.out.println(clientes);
    }

    public static void Q2(){
        Map<String, Integer> produtos = new HashMap<>(Map.of(
            "teclado", 50,
            "mouse", 120,
            "monitor", 30
        ));
        while(true){
            System.out.print("Digite o produto à pesquisar ou quit para saír: ");
            String produto = scanner.nextLine().toLowerCase();
            if("quit".equals(produto)) break;
            if(produtos.containsKey(produto)){
                System.out.println("Quantidade do " + produto + ": " + produtos.get(produto));
            }else System.out.println("Produto não existe");

        }

    }

    public static void Q3(){
        Map<String, Double> faturamento_por_regiao = new HashMap<>(Map.of(
            "Norte", 15000.00,
            "Sul", 22000.00,
            "Leste", 18000.00,
            "Oeste", 25000.00
        ));
        List<Double> faturamentos = new ArrayList<>();
        for (Map.Entry<String, Double> entrada : faturamento_por_regiao.entrySet()) {
            faturamentos.add(entrada.getValue());
        }
        System.out.println("Faturamentos: " + faturamentos);
        Double faturamento_total = 0.0;
        for(double faturamento: faturamentos){
            faturamento_total += faturamento;
        }
        System.out.printf("Faturamento Total: R$%.2f\n", faturamento_total);
        System.out.printf("Faturamento Médio: R$%.2f\n", (faturamento_total/faturamentos.size()));

    }

    public static void Q4(){
        Map<String, List<Integer>> notas = new HashMap<>(Map.of(
            "Lira", new ArrayList<>(Arrays.asList(8, 9, 7)),
            "Paula", new ArrayList<>(Arrays.asList(10, 9, 10)),
            "Tiago", new ArrayList<>(Arrays.asList(6, 7, 8))
        ));
        String key = "Paula";
        if(notas.containsKey(key)){
            double soma_notas = 0.0;
            List<Integer> boletim = notas.get(key);
            for(double nota:boletim){
                soma_notas += (double)nota;
            }
            System.out.printf("A média da %s é de %.1f.\n", key, (soma_notas/notas.size()));
        }
    }

    public static void Q5(){
        Map<String, Double> produtos = new HashMap<>(Map.of(
            "celular", 1500.00,
            "camera", 800.00,
            "radio", 200.00,
            "fone", 100.00
        ));
        String key = "radio";
        Double produto_removido = produtos.remove(key);
        if(produto_removido != null) System.out.printf("Produto Removido: %s (valor: R$%.2f).\n", key, produto_removido);
        else System.out.println("O produto não existe");
        String key2 = "celular";
        System.out.println("Celular ainda existe: " + produtos.containsKey(key2));
        //System.out.println(produtos);

    }
    public static void main(String[] args) {
        System.out.println("Atualização de Cadastro de Clientes (Setor de CRM)");
        Q1();
        System.out.println("Consulta de Estoque Interativa (Setor de Logística)");
        Q2();
        System.out.println("Análise de Faturamento por Região (Setor Financeiro)");
        Q3();
        System.out.println("Sistema de RH – Média de Desempenho (Setor de RH)");
        Q4();
        System.out.println("Limpeza de Banco de Dados (Setor de TI)");
        Q5();
    }
    
}

File: My Answer/E6.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>
#include <iomanip>
#include <algorithm>
#include <cctype>
#include <locale>
#include <regex>
#include <cmath>
#include <stdexcept>
#include "utils.h"

using namespace std;

void Q1(){
    map<string, double> clientes = {
        {"Lira", 5000.0},
        {"Alon", 3000.0},
        {"Julia", 4500.0}
    };

    clientes["Alon"] += 1500.00;

    clientes["Marcos"] = 2000.00;

    cout << "--- Cadastro de Clientes Atualizado ---" << endl;
    
    for (const auto& registro : clientes) {
        cout << "Cliente: " << registro.first << " | Faturamento: R$" << registro.second << endl;
    }
    cout << "---------------------------------------" << endl;
}

void Q2(){
    map<string, float> estoque = {
        {"teclado", 50},
        {"mouse", 20},
        {"monitor", 30}
    };

    string produto;
    getline(cin, produto);
    replaceAll(produto, " ", "");
    transform(produto.begin(), produto.end(), produto.begin(), ::tolower);
    bool flag = (estoque.find(produto) != estoque.end());
    if(flag) cout << "Quantidade do " << produto << ": " << estoque[produto] << endl;
    else cout << "Produto não encontrado no sistema.\n";
}


void Q3(){
    float sum = 0;
    map<string, float> vendas_regiao = {
        {"Norte", 15000},
        {"Sul", 22000},
        {"Leste", 18000},
        {"Oeste", 25000}
    };
    for(const auto& regiao : vendas_regiao){
        cout << "Vendas na região " << regiao.first << ": R$" << fixed << setprecision(2) << regiao.second << endl;
        sum += regiao.second;
    }
    cout << "Total de vendas: R$" << fixed << setprecision(2) << sum << endl;
    cout << "Média de vendas por região: R$" << fixed << setprecision(2) << (sum / vendas_regiao.size()) << endl;
}

void Q4(){
    map<string, vector<float>> desemmpenho = {
        {"Lira", {8, 9, 7}},
        {"Paula", {10, 9, 10}},
        {"Tiago", {6, 7, 8}}
    };
    vector<float> lista_notas = desemmpenho["Paula"];
    float media = 0;
    for(float nota : lista_notas){
        media += nota;
    }
    cout << "A média da paula foi: " << fixed << setprecision(2) << (media / lista_notas.size()) << endl;
}

void Q5(){
    map<string, float> produtos = {
        {"celular", 1500},
        {"camera", 800},
        {"radio", 200},
        {"fone", 100}
    };
    // remover rádio com o equivalente ao .pop(), imprimir o valor removido, verificar se o celular ainda existe.
    float valor_removido = produtos["radio"];
    produtos.erase("radio");
    cout << "Valor do produto removido (radio): R$" << fixed << setprecision(2) << valor_removido << endl;
    bool celular_existe = (produtos.find("celular") != produtos.end());
    cout << "O produto 'celular' " << (celular_existe ? "existe no estoque." : "não existe no estoque.") << endl;
    cout << (celular_existe ? "True" : "False") << endl;
}

int main(){
    //Q1();
    //Q2();
    Q3();
    cout<<endl;
    Q4();
    Q5();
    return 0;
}